DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs

```diff
@@ -349,6 +349,41 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
                 capturedException.Throw();
             }
 
+            if (repositoryDevice != null && repositoryDevice.IsSimulatedDevice)
```

Comment:
can you add some unit tests for this part? If you wish you can put this code in a public method called something like: AddOrRemoveSimulatedDevice() and test that method